lib/vanagon/component.rb

```diff
@@ -290,6 +290,10 @@ def get_build_dir
       end
     end
 
+    def get_dependency_hash
+      { name => { 'version' => version, 'ref' => options[:ref] }.delete_if { |k, v| !v } }
```

Comment:
I think i'm going to leave this as is. I do not want to get into the lower level implementation details between what is the best way of creating a simple hash like this. If we have enough components that choosing between the different ways to create a hash will adversely affect performance then there are probably bigger and better problems to solve before getting to this one.